src/components/VideoEditor.tsx

```diff
@@ -0,0 +1,229 @@
+    let newChannel: Channel;
+    if (key === "name") {
+      newChannel = {
+        link: vidData?.channel?.link || "",
+        name: event.currentTarget.value
+      }
+    } else {
+      newChannel = {
+        link: event.currentTarget.value,
+        name: vidData?.channel?.name || "",
+      }
+    }
+
+    const newVidData: Partial<Ambiance> = {
+      ...vidData,
+      channel: newChannel
+    }
+    setVidData(newVidData);
+  }
+
+
+  return (
+    <div id='Video-Editor'>
+      <Stack className='text-field-stack' direction="column" spacing={.4}>
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.name}
+          onChange={(e) => handleOnChange(e, "name")}
+          helperText="Display Name"
+          size='small'
+          variant="filled"
+          required
+          value={vidData?.name || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.group}
+          onChange={(e) => handleOnChange(e, "group")}
+          helperText="Subcategory"
+          variant="filled"
+          size='small'
+          required
+          value={vidData?.group || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.code}
+          onChange={(e) => handleOnChange(e, "code")}
+          helperText="YouTube URL"
+          variant="filled"
+          size='small'
+          required
+          value={vidData?.code || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.channel?.name}
+          onChange={(e) => handleOnChangeChannel(e, "name")}
+          helperText="Optional: YouTube channel Name"
+          variant="filled"
+          size='small'
+          value={vidData?.channel?.name || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.channel?.link}
+          onChange={(e) => handleOnChangeChannel(e, "link")}
+          helperText="Optional: YouTube channel Link"
+          variant="filled"
+          size='small'
+          value={vidData?.channel?.link || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.startTimeS}
+          onChange={(e) => handleOnChange(e, "startTimeS")}
+          helperText="Optional: Video start time (seconds)"
+          variant="filled"
+          size='small'
+          value={vidData?.startTimeS || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.livestream ? "Yes" : "No"}
+          onChange={(e) => handleOnChange(e, "livestream")}
+          helperText="Optional: Yes / No livestream"
+          variant="filled"
+          size='small'
+          select
+          value={vidData?.livestream || ""}
+        >
```

Comment:
This field is causing a bug in our DB as its being stored as a String when it should be a boolean. We will need to convert the value set here into a boolean before saving to DB.

Reply:
Example:
![Screenshot 2024-04-05 at 10 48 15 AM](https://github.com/hermanator608/ambiance/assets/29784945/64a79650-ec57-45da-bbff-9fbf7e9c7d1b)
![Screenshot 2024-04-05 at 10 48 08 AM](https://github.com/hermanator608/ambiance/assets/29784945/51434585-2c1a-405c-aa87-fc2af80ea340)